### PR TITLE
Octree

### LINKED ISSUE #1435: Octree data structure in C++

### PR DESCRIPTION
**Pull request by dldisha**
Fixes #1435 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Octree data structure added in C++

#### Languages Used:
- C++

#### Files Added:
- Octree/Octree.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Octree/Octree.cpp [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+#define BLB 7    // bottom left back
+
+struct Point
+{
+    int x;
+    int y;
+    int z;
+
+    Point() : x(-1), y(-1), z(-1) {}
+
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Remove this line.

File: Octree/Octree.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            {
+                if (z <= midz)
+                    pos = BRF;
+                else
+                    pos = BRB;
+            }
+        }
+        if (children[pos] -> point == nullptr)
+        {
+           // if region node
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Fix indentation.

File: Octree/Octree.cpp [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        {
+            int x_ = children[pos] -> point -> x,
+                y_ = children[pos] -> point -> y,
+                z_ = children[pos] -> point -> z;
+            delete children[pos];
+            children[pos] = nullptr;
+            if(pos == TLF)
+            {
+                children[pos] = new Octree(top_left_front -> x, top_left_front -> y, top_left_front -> z,
+                                        midx, midy, midz);
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Fix indentation here. 16-space indentation would do I believe.

File: Octree/Octree.cpp [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+            }
+            else if(pos == TRF)
+            {
+                children[pos] = new Octree(midx + 1, top_left_front -> y, top_left_front -> z,
+                                        bottom_right_back -> x, midy, midz);
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Here as well, indentation.

File: Octree/Octree.cpp [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+            }
+            else if(pos == BRF)
+            {
+                children[pos] = new Octree(midx + 1, midy + 1, top_left_front -> z,
+                                        bottom_right_back -> x, bottom_right_back -> y, midz);
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Indentation in all such places.

File: Octree/Octree.cpp [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+            return;
+        }
+        else
+        {
+            int x_ = children[pos] -> point -> x,
+                y_ = children[pos] -> point -> y,
+                z_ = children[pos] -> point -> z;
+            delete children[pos];
+            children[pos] = nullptr;
+            if(pos == TLF)
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Space after if.

File: Octree/Octree.cpp [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+            {
+                children[pos] = new Octree(top_left_front -> x, top_left_front -> y, top_left_front -> z,
+                                        midx, midy, midz);
+            }
+            else if(pos == TRF)
```
Comment (marwahmanbir) [inline Octree/Octree.cpp]: Space after if at all places.

File: Octree/README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
+# Octree Data Structure
```
Comment (marwahmanbir) [inline Octree/README.md]: This file looks good.

File: Skip_list/README.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
+# SKIP LIST
```
Comment (marwahmanbir) [inline Skip_list/README.md]: I guess you have added Skip list README.md and skip_list.cpp by mistake as these files do not seem to be related to this PR. Please fix this.

**Comment by MastersAbh** [comment]
@marwahmanbir please have a look at it.

**Comment by dldisha** [comment]
@jainaman224 @rishabhgarg25699, please close and merge this PR.
Thank you
